rust_src/src/eval.rs

```diff
@@ -460,6 +460,23 @@ pub fn lisp_let(args: LispCons) -> LispObject {
     LispObject::from_raw(unsafe { unbind_to(count, val) })
 }
 
+/// If TEST yields non-nil, eval BODY... and repeat.
+/// The order of execution is thus TEST, BODY, TEST, BODY and so on
+/// until TEST returns nil.
+/// usage: (while TEST BODY...)
+#[lisp_fn(name = "while", c_name = "while", min = "1", unevalled = "true")]
+pub fn lisp_while(args: LispCons) -> LispObject {
+    let (test, body) = args.as_tuple();
+
+    while unsafe { eval_sub(test.to_raw()) } != Qnil {
+        unsafe { maybe_quit() };
+
```

Comment:
I'm not a fan of having lots of blank lines in things, but this one here is the silliest.

Reply:
style difference. whitespace exists to make code easier to read. `maybe_quit` and `progn` body have nothing to do with each other so they are broke up.